statsmodels/base/model.py

```diff
@@ -797,22 +797,24 @@ def predict(self, exog=None, transform=True, *args, **kwargs):
         """
         import pandas as pd
 
-        exog_index = exog.index if _is_using_pandas(exog, None) else None
+        is_pandas = _is_using_pandas(exog, None)
 
-        if transform and hasattr(self.model, 'formula') and exog is not None:
+        exog_index = exog.index if is_pandas else None
+
+        if transform and hasattr(self.model, 'formula') and (exog is not None):
             from patsy import dmatrix
-            exog = pd.DataFrame(exog)  # user may pass series, if one predictor
-            if exog_index is None:  # user passed in a dictionary
-                exog_index = exog.index
-            exog = dmatrix(self.model.data.design_info.builder,
+            if isinstance(exog, pd.Series):
+                exog = pd.DataFrame(exog)
```

Comment:
the problem here is that we don't have the information to tell whether this should be transposed, i.e. do we have one row with many exog, or one exog with many rows.

My guess is that the former case would be more common, i.e. needs a  `.T`.
The user can provide a DataFrame instead of a Series to make it unambiguous.